src/main/java/org/spongepowered/api/command/args/GenericArguments.java

```diff
@@ -631,31 +634,26 @@ public static CommandElement optionalWeak(CommandElement element, Object value)
 
         @Override
         public void parse(CommandSource source, CommandArgs args, CommandContext context) throws ArgumentParseException {
-            if (!args.hasNext()) {
-                Text key = this.element.getKey();
-                if (key != null && this.value != null) {
-                    context.putArg(key.toPlain(), this.value);
-                }
-                return;
-            }
-            Object startState = args.getState();
-            try {
-                this.element.parse(source, args, context);
-            } catch (ArgumentParseException ex) {
-                if (this.considerInvalidFormatEmpty || args.hasNext()) { // If there are more args, suppress. Otherwise, throw the error
-                    args.setState(startState);
-                    if (this.element.getKey() != null && this.value != null) {
-                        context.putArg(this.element.getUntranslatedKey(), this.value);
+            if (args.hasNext()) {
+                Object start = args.getState();
+                try {
+                    this.element.parse(source, args, context);
+                } catch (ArgumentParseException ex) {
+                    if (!this.considerInvalidFormatEmpty && !args.hasNext()) {
+                        throw ex;
                     }
-                } else {
-                    throw ex;
+                    args.setState(start);
                 }
             }
+            if (this.element.getKey() != null && this.value != null) {
```

Comment:
Because there is no return after the parsing, you're now _always_ adding the fallback value to the context, regardless of whether you've parsed something successfully or not.

Reply:
Good catch!